server/src/client_service.rs

```diff
@@ -566,64 +580,69 @@ mod tests {
         let quotes = quote_book.get_quotes(&pair, .., 0).unwrap();
         assert_eq!(quotes.len(), 1);
 
-        // Remove it.
-        let quote_id = deqs_api::deqs::QuoteId::from(quotes[0].id());
-        let mut req = RemoveQuoteRequest::default();
-        req.set_quote_id(quote_id);
-
-        let resp = client_api.remove_quote(&req).expect("remove quote failed");
-        assert_eq!(resp.get_quote(), &(&quotes[0]).into());
-
-        // Try again, should fail.
-        assert!(
-            matches!(client_api.remove_quote(&req), Err(grpcio::Error::RpcFailure(status)) if status.code() == grpcio::RpcStatusCode::NOT_FOUND)
-        );
+        let added_quote = &resp.get_quotes()[0];
 
-        // Invalid quote id should fail.
-        let req = RemoveQuoteRequest::default();
-        assert!(
-            matches!(client_api.remove_quote(&req), Err(grpcio::Error::RpcFailure(status)) if status.code() == grpcio::RpcStatusCode::INVALID_ARGUMENT)
-        );
+        // We should now see our quote arrive in the stream.
+        let resp = live_updates_rx.recv().expect("recv failed");
+        assert_eq!(resp.get_quote_added(), added_quote);
     }
 
     #[test_with_logger]
-    fn streaming_works_without_filtering(logger: Logger) {
+    fn streaming_works_with_quotes_expiring(logger: Logger) {
         let pair = Pair {
             base_token_id: TokenId::from(0),
             counter_token_id: TokenId::from(1),
         };
         let mut rng: StdRng = SeedableRng::from_seed([1u8; 32]);
-        let quote_book = InMemoryQuoteBook::default();
-        let (client_api, _server, _msg_bus_rx) =
-            create_test_client_and_server(&quote_book, &logger);
 
-        let (tx, rx) = channel();
+        let mut ledger = create_and_initialize_test_ledger();
+        let starting_blocks = ledger.num_blocks().unwrap();
+
+        // Allow msg_bus_tx to be created later for remove_quote_callback,
+        // by putting Mutex<Option<Sender>> in the captured state of the callback
+        let remove_quote_callback_state = Arc::new(Mutex::new(Option::<Sender<Msg>>::default()));
```

Comment:
this is the trick i made up to work around the issue of server and quotebook both needing access to msg bus, without changing the other parts of the test infra,

it seems okay to me, but lmk if you think we should do this differently

Reply:
Well I suppose if we are already simulating the delivery on the message bus, maybe you can skip the SynchronizedQuoteBook and just post directly to the message bus and see that it shows in the live update stream?

Reply:
yeah i guess if this module is supposed to be testing the client api service, we can focus on just testing specifically that, and send the cancellation notice to `msg_bus_tx`. okay will fix